### PR TITLE
Borders: Update border color for multiple components

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Update Form Fields and Tables to use new border design Token. The main noticeable differences are in Switch, Checkbox, and RadioButton. 

#### Switch

Before
Screen Shot 2022-03-29 at 1 51 36 PM

After
Screen Shot 2022-03-29 at 1 54 17 PM

#### Checkbox

Before
Screen Shot 2022-03-29 at 1 57 43 PM

After
Screen Shot 2022-03-29 at 1 57 55 PM

#### RadioButton

Before

Screen Shot 2022-03-29 at 1 58 11 PM

After
Screen Shot 2022-03-29 at 1 58 25 PM

#### Why?

In order to create consistent borders that are more accessible, this PR updates our form fields and tables to use our border design token colors. The main change in Controls components is designed to help out switches look more intentional, and to make disabled switches look more disabled. It also improves the accessibility because our previous color was not accessible against either our light ro dark backgrounds.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4080)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: packages/gestalt/src/Switch.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export default function Switch({
   const sliderStyles = classnames(
     styles.slider,
     switched ? styles.sliderRight : styles.sliderLeft,
-    switched && !disabled ? styles.sliderDark : styles.sliderLight,
+    !disabled ? styles.sliderDark : styles.sliderLight,
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Switch.js:74]: Intentional change to help our default switch look active, and our disabled switch to look more disabled

**Comment by AlbertCarreras** [comment]
Intended?
Screen Shot 2022-03-29 at 1 49 37 PM

**Comment by AlbertCarreras** [comment]
These changes should have triggered the vis diff Integration test to fail... I wonder how it did pass it.

**Comment by ayeshakmaz** [comment]
@AlbertCarreras made some updates and fixed that issue - thanks for taking a look!

**Comment by ponori** [comment]
Hey Ayesha - they may be a discrepancy between design and prod related to our disabled state. Just noting this as something we should align on in the future - it shouldn't block this PR.

File: packages/gestalt/src/Switch.js (hunk 1 above)
Comment (dangerismycat) [reply packages/gestalt/src/Switch.js:74]: They still don't read as "disabled" enough to me, but this is an improvement! 😅 And that's beyond the scope of this PR

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**